src/libAtomVM/nifs.c

```diff
@@ -3272,6 +3279,88 @@ static term nif_binary_split(Context *ctx, int argc, term argv[])
     return result_list;
 }
 
+static term nif_binary_replace(Context *ctx, int argc, term argv[])
+{
+    term bin_term = argv[0];
+    term pattern = argv[1];
+    term replacement = argv[2];
+    term options = argc == 4 ? argv[3] : term_nil();
+
+    VALIDATE_VALUE(bin_term, term_is_binary);
+    VALIDATE_VALUE(pattern, term_is_binary);
+    VALIDATE_VALUE(replacement, term_is_binary);
+    VALIDATE_VALUE(options, term_is_list);
+
+    bool global = interop_proplist_get_value_default(options, GLOBAL_ATOM, FALSE_ATOM) == TRUE_ATOM;
```

Comment:
I suggest:
```
bool global;
switch (interop_proplist_get_value_default(options, GLOBAL_ATOM, FALSE_ATOM)) {
   case TRUE_ATOM:
...
   case FALSE_ATOM:
...
   default:
      raise badarg here
```

I think the OTP one raises if global option has some weird value that is not an atom.
We might add `term_is_boolean()` helper in term.h by the way, in that case let's add that in an additional commit.
So we can add `VALIDATE_VALUE(global_term, term_is_boolean)`

Reply:
I replaced it with the loop. `term_is_boolean` isn't necessary IMO, you'd need to distinguish `{opt, true}` and `opt` cases (where latter's existence sets to true). Maybe some more general option parser would be nice (e.g. `interop_get_options(options_term, &options_struct)`).